### ISSUE TITLE
[SPIKE] Investigate getting custom data into Intake

### ISSUE THREAD

**Issue by jacobthill** [sul-dlss/dlme#1296]
The idea of moving all incoming data into a Pandas DataFrame (DF) has some advantages (listed in https://github.com/sul-dlss/dlme-harvest/issues/87). One open question is how complicated it will be to get the heterogeneous data coming form data providers into a DF using Intake. The data in DLME is in the following formats:

- Many json records, mostly from iiif harvests:
   - https://github.com/sul-dlss/dlme-metadata/tree/main/loc
   - https://github.com/sul-dlss/dlme-metadata/tree/main/shahre-farang
   - https://github.com/sul-dlss/dlme-metadata/tree/main/bodleian
- Many xml files:
   - https://github.com/sul-dlss/dlme-metadata/tree/main/bnf
   - https://github.com/sul-dlss/dlme-metadata/tree/main/cambridge/genizah-1
   - https://github.com/sul-dlss/dlme-metadata/tree/main/qnl
- Some csv files (I assume this will be straightforward to get into a Pandas DF with Intake):
   - https://github.com/sul-dlss/dlme-metadata/tree/main/yale/babylonian/data
   - https://github.com/sul-dlss/dlme-metadata/blob/main/penn/egyptian-museum/data/egyptian-20181028-cleaned.csv

We need to get all values in the incoming data into the DF. The json is typically pretty simple but want to make sure nested dicts are converted properly to a DF. I assume the csv will be straightforward. The XML is what gives me the most concern. We need to map nested XML nodes but also attributes and values from the XML in a format that is clear:

- If a record has a node with a language value stored as an attribute values of that node (e.g. https://github.com/sul-dlss/dlme-metadata/blob/main/penn/openn/genizah-0002/data/0002-1.xml)
- How does it do with MARC XML, particularly the `subfield` nodes and all of the confusing `code`, `tag`, and `ind*` attributes and subfields (e.g. https://raw.githubusercontent.com/sul-dlss/dlme-metadata/main/michigan/data/michigan-1001.xml)

Basically I think the question I have is can we come up with an abstract way of naming DF columns from XML data (e.g. Would the name need to be something like `top-level-node_child-node_grand-child-node_attribute_value` to capture the information needed to make sense of the data and map it?)

**Comment by jermnelson** [comment]
I created a branch `df-intake-spike` in the dlme-metadata repository with a Jupyter notebook, `dataframes.ipynb` that contains my analysis so far on this ticket. You can use MyBinder to launch and view the notebook at https://mybinder.org/v2/gh/sul-dlss/dlme-metadata/df-intake-spike.
